apache/ozone
# HDDS-6235. Empty KeyValueContainers are replicated without chunks directory.

## What changes were proposed in this pull request?

- Include empty chunks directory in the tarball when replicating a container.
- Recreate directory entries when unpacking the tarball.
- On datanode startup, create an empty chunks directory if no chunks directory is present.
    - Repairs existing containers now that the bug is fixed.

## What is the link to the Apache JIRA

HDDS-6235

## How was this patch tested?

- Unit test added, which passed only after the fix was applied.
- Manually tested chunks directory creation on startup in docker:
    1. Create empty container with `ozone admin container create`
    2. Delete its chunks directory.
    3. Restart the datanode.
    4. Check that empty chunks directory is created.
